src/DynamoCore/Library/LibraryServices.cs

```diff
@@ -671,6 +671,48 @@ private void PopulatePreloadLibraries()
 
         }
 
+        /// <summary>
+        /// Try get default argument expression from DefaultArgumentAttribute, 
+        /// and parse into Associaitve AST node. 
+        /// </summary>
+        /// <param name="arg"></param>
+        /// <param name="defaultArgumentNode"></param>
+        /// <returns></returns>
+        private bool TryGetDefaultArgumentFromAttribute(ArgumentInfo arg, out AssociativeNode defaultArgumentNode)
+        {
+            defaultArgumentNode = null;
+
+            if (arg.Attributes == null)
+                return false;
+
+            object o;
+            if (!arg.Attributes.TryGetAttribute("DefaultArgumentAttribute", out o))
+                return false;
+
+            var defaultExpression = o as string;
+            if (string.IsNullOrEmpty(defaultExpression))
+                return false;
+
+            var currentParsingmode = LibraryManagementCore.ParsingMode;
+            var currentParsingFlag = LibraryManagementCore.IsParsingCodeBlockNode;
+
+            LibraryManagementCore.ParsingMode = ProtoCore.ParseMode.AllowNonAssignment;
+            LibraryManagementCore.IsParsingCodeBlockNode = true;
+
+            var astNode = ParserUtils.ParseWithCore(defaultExpression + ";", LibraryManagementCore);
+            if (astNode != null)
+            {
+                var cbn = astNode as CodeBlockNode;
+                if (cbn != null && cbn.Body.Any())
+                    defaultArgumentNode = (cbn.Body[0] as BinaryExpressionNode).RightNode;
+            }
+
+            LibraryManagementCore.ParsingMode = currentParsingmode;
```

Comment:
You are right, [next subtask](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-6507) is exactly for error handling in default argument expression.